crates/aranya-daemon/src/api.rs

```
@@ -351,24 +385,105 @@ impl DaemonApi for Api {
+                            crypto
+                                .aranya_store
+                                .get_key(&mut crypto.engine, enc_id.into_id())
+                                .context("keystore error")?
+                                .context("missing enc_sk in add_team")?
+                        };
+
+                        let group = GroupId::from(team.into_id());
+                        let seed = enc_sk
+                            .open_psk_seed(
+                                &wrapped.encap_key,
+                                wrapped.encrypted_seed,
+                                &wrapped.sender_pk,
+                                &group,
+                            )
+                            .context("could not open psk seed")?;
+                        qs::PskSeed(seed)
+                    }
+                };
+
+                self.add_seed(team, seed.clone()).await?;
+
+                for psk_res in seed.generate_psks(team) {
+                    let psk = psk_res.context("unable to generate psk")?;
+                    psk_store
+                        .insert(team, Arc::new(psk))
+                        .inspect_err(|err| error!(err = ?err, "unable to insert PSK"))?
+                }
+
+                Ok(())
+            }
+            None => Err(anyhow!("Missing QUIC sync config").into()),
+        }
     }
 
     #[instrument(skip(self))]
     async fn remove_team(self, _: context::Context, team: api::TeamId) -> api::Result<()> {
-        todo!();
+        let mut errors = vec![];
+        if let Some(data) = &self.quic {
+            let _ = data
+                .psk_store
+                .remove(team)
+                .inspect_err(|err| error!(err = ?err, "unable to remove PSK"))
+                .map_err(|err| errors.push(err));
+        }
+
+        let _ = self
+            .seed_id_dir
+            .remove(&team)
+            .await
+            .map_err(|err| errors.push(err));
```

If we don't want to return an error if the PSK removal fails here, let's at least log a `warn!`.
```suggestion
        if let Err(e) = self
            .seed_id_dir
            .remove(&team)
            .await
            .map_err(|err| errors.push(err)) {
                warn!(e);
            }
```